### DISCUSSION TITLE
Heading Permalinks + aria-hidden

### DISCUSSION THREAD

**Discussion by joshbruce** [thephpleague/commonmark#738] [Q&A] [answered]
Was doing some analytics using web.dev using the following page: https://joshbruce.com/web-development/2021-site-in-depth

Received the following feedback related to the HTML rendered by the heading permalink extension (reduces accessibility score):

<img width="931" alt="Screen Shot 2021-11-03 at 1 14 12 PM" src="https://user-images.githubusercontent.com/15252830/140168599-c035c908-96cf-4110-8099-0b98a4b67905.png">

Rendered output:

<img width="922" alt="Screen Shot 2021-11-03 at 1 10 12 PM" src="https://user-images.githubusercontent.com/15252830/140168786-a720b95e-f079-4af0-971a-d0082bcce65a.png">

Alternative seen in the wild (non-GitHub, because they also use aria-hidden):

<img width="474" alt="Screen Shot 2021-11-03 at 1 08 49 PM" src="https://user-images.githubusercontent.com/15252830/140169016-8f43cf5a-84fb-4ff2-b980-6dfcbe25b662.png">

GitHub template also gets hit for accessibility on this score:

<img width="936" alt="Screen Shot 2021-11-03 at 1 17 59 PM" src="https://user-images.githubusercontent.com/15252830/140169183-8d3ae036-84c9-4b15-8044-4cd060b32a4d.png">

As I was writing this I thought about some possibilities...

1. Make the aria-hidden optional in the extension - with true as default - could be the most direct and backward compatible way; I would be willing to make that contribution.
2. Removing it entirely would probably break the GFM because GitHub does use aria-hidden.

Thanks as always!

**Comment by colinodell** [comment, marked as answer]
Great find!  Yeah, I'd prefer to keep close alignment to GFM as much as possible, but I think this change makes a ton of sense.

**Comment by colinodell** [comment]
I tried to do more research on this subject but the only relevant info I could find was this: https://amberwilson.co.uk/blog/are-your-anchor-links-accessible/

**Comment by joshbruce** [comment]
Went to check the USWDS and see what they're thinking - they link to the same article along with a brief discussion: https://github.com/uswds/uswds/issues/4166

Put it on my radar: https://github.com/8fold/site-joshbruce.com/issues/72

It's an interesting problem and Amber summed it up well and seems more specialized than me. The option of turning off `aria-hidden` should help (thanks for the merge) and I'm thinking of extending and extension - that literally possible?? lol

**Comment by joshbruce** [comment]
Just did a quick verification on Amber's site while creating the extension and it passes despite the `aria-hidden` because it has no focusable descendants.

<img width="928" alt="Screen Shot 2021-11-12 at 9 31 03 PM" src="https://user-images.githubusercontent.com/15252830/141604263-d059aff5-2545-40c2-886a-1fc8f827d759.png">
